<!-- Launch file for Movo teleop startup -->

<launch>
    <arg name="sim" default="false"/>
    <arg name="sim_teleop_vel_limit_mps" default="0.5"/>
    <arg name="sim_teleop_yaw_rate_limit_rps" default="0.5"/>
    <arg name="sim_teleop_accel_lim" default="0.5"/>
    <arg name="sim_teleop_yaw_accel_lim" default="0.5"/>
    <arg name="xbox_joy" default="$(optenv MOVO_JOY_IS_XBOX)" /> 



    <node pkg="joy" type="joy_node" name="joy_node" if="$(optenv MOVO_HAS_ONBOARD_JOY false)">
        <param name="dev" type="string" value="$(optenv MOVO_JOY_DEV /dev/input/js0)" />
        <param name="deadzone" type="double" value="0.1" />
        <param name="autorepeat_rate" type="double" value="50" />
    </node>
    
    <group unless="$(arg xbox_joy)"> 
        <node pkg="movo_ros" type="movo_teleop" name="movo_teleop" output="screen">
            <param name="sim" value="$(arg sim)"/>
            <param name="sim_teleop_vel_limit_mps" value="$(arg sim_teleop_vel_limit_mps)"/>
            <param name="sim_teleop_yaw_rate_limit_rps" value="$(arg sim_teleop_yaw_rate_limit_rps)"/>
            <param name="sim_teleop_accel_lim" value="$(arg sim_teleop_accel_lim)"/>
            <param name="sim_teleop_yaw_accel_lim" value="$(arg sim_teleop_yaw_accel_lim)"/>
        </node>
    </group>
    <group if="$(arg xbox_joy)"> 
        <node pkg="movo_ros" type="movo_teleop_xbox" name="movo_teleop" output="screen">
            <param name="sim" value="$(arg sim)"/>
            <param name="sim_teleop_vel_limit_mps" value="$(arg sim_teleop_vel_limit_mps)"/>
            <param name="sim_teleop_yaw_rate_limit_rps" value="$(arg sim_teleop_yaw_rate_limit_rps)"/>
            <param name="sim_teleop_accel_lim" value="$(arg sim_teleop_accel_lim)"/>
            <param name="sim_teleop_yaw_accel_lim" value="$(arg sim_teleop_yaw_accel_lim)"/>
        </node>
    </group>
    
    <node pkg="movo_ros" type="movo_marker_ctrl" name="movo_marker_ctrl" output="screen">
        <param name="sim" value="$(arg sim)"/>
    </node>
    
    <group ns="movo">
        <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager"/>
        <node pkg="nodelet" type="nodelet" name="cmd_vel_mux" args="load yocs_cmd_vel_mux/CmdVelMuxNodelet nodelet_manager">
            <param name="yaml_cfg_file" value="$(find movo_ros)/launch/config/cmd_vel_mux.yaml"/>
        </node>
    </group>
</launch> 
